ports/stm32/rfcore.c

```diff
@@ -589,18 +600,38 @@ static const struct {
 void rfcore_ble_init(void) {
     DEBUG_printf("rfcore_ble_init\n");
 
-    // Clear any outstanding messages from ipcc_init.
-    tl_check_msg(&ipcc_mem_sys_queue, IPCC_CH_SYS, NULL);
-
     // Configure and reset the BLE controller.
-    tl_sys_hci_cmd_resp(HCI_OPCODE(OGF_VENDOR, OCF_BLE_INIT), (const uint8_t *)&ble_init_params, sizeof(ble_init_params), 0);
-    tl_ble_hci_cmd_resp(HCI_OPCODE(0x03, 0x0003), NULL, 0);
+    if (!rfcore_ble_reset()) {
+        // ble init can fail if core2 has previously locked up. Reset HSI & rfcore to retry.
+        __HAL_RCC_HSI_DISABLE();
+        mp_hal_delay_ms(100);
+        __HAL_RCC_HSI_ENABLE();
```

Comment:
maybe use `LL_RCC_HSI_Enable()` etc, to be consistent with the new code added in `powerctrl.c`

Reply:
Yeah... sorry I did think I should check the implementation of both of them to confirm if the HAL variant does more useful stuff over the LL one to select which is actually most appropriate here. These HAL ones are used elsewhere already too...

Reply:
The LL ones are fine to use.  I think they are exactly what's needed and nothing more.

Reply:
Oh hah, well the stm impl for the hal functions doesn't add much...
```
#define __HAL_RCC_HSI_ENABLE()  LL_RCC_HSI_Enable()
#define __HAL_RCC_HSI_DISABLE() LL_RCC_HSI_Disable()
```